eessi_container.sh

```diff
@@ -79,7 +79,7 @@ display_help() {
   echo "  -m | --mode MODE       - with MODE==shell (launch interactive shell) or"
   echo "                           MODE==run (run a script or command) [default: shell]"
   echo "  -r | --repository CFG  - configuration file or identifier defining the"
-  echo "                           repository to use [default: EESSI-pilot via"
+  echo "                           repository to use [default: EESSI via"
   echo "                           default container, see --container]"
```

Comment:
The container images we use should probably be updated with the new configuration too.

Reply:
Well, yes and no.

Once https://github.com/cvmfs-contrib/config-repo/pull/206 is merged, `software.eessi.io` should become automatically available with the current build (and client) container, I think.

So I prefer to try and see if that's the case first before we update the build container (which may be wise anyway, to sync up with the latest CernVM-FS for example).

Reply:
If we want to make use of the configuration provided by `cvmfs-config-default`, we do have to make sure to also mount `/cvmfs/cvmfs-config.cern.ch` in the container, as that provides the configuration files. So, we need to add a `--fusemount container:cvmfs2 cvmfs-config.cern.ch /cvmfs/cvmfs-config.cern.ch` to the `apptainer` command.

Reply:
When trying interactively, for some reason accessing `/cvmfs/cvmfs-config.cern.ch` just works even without the extra `--fusemount` option, and hence so does accessing `/cvmfs/software.eessi.io`:

```
Apptainer> find /etc/cvmfs | grep eessi
/etc/cvmfs/domain.d/eessi-hpc.org.conf
/etc/cvmfs/keys/eessi-hpc.org
/etc/cvmfs/keys/eessi-hpc.org/pilot.eessi-hpc.org.pub
/etc/cvmfs/keys/eessi-hpc.org/cvmfs-config.eessi-hpc.org.pub
/etc/cvmfs/keys/eessi-hpc.org/ci.eessi-hpc.org.pub

Apptainer> find /etc/cvmfs | grep eessi.io
Apptainer>

Apptainer> ls /cvmfs/cvmfs-config.cern.ch
etc


Apptainer> find /cvmfs/cvmfs-config.cern.ch -type f  grep eessi.io | xargs sha256sum
find: paths must precede expression: `grep'
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855  -
Apptainer> find /cvmfs/cvmfs-config.cern.ch -type f | grep eessi.io | xargs sha256sum
7d484216c59c00b81c9eb89e5ac511930d0e601a36cc2bdb106f51d148b19f14  /cvmfs/cvmfs-config.cern.ch/etc/cvmfs/domain.d/eessi.io.conf
d86b2190859f5124e78e0ff711c5a042be10d569350130a0f10a537eaacbc86b  /cvmfs/cvmfs-config.cern.ch/etc/cvmfs/keys/eessi.io/eessi.io.pub

Apptainer>
Apptainer> ls /cvmfs/software.eessi.io
host_injections  versions

Apptainer> attr -g revision /cvmfs/software.eessi.io
Attribute "revision" had a 2 byte value for /cvmfs/software.eessi.io:
25
```

Maybe loading one CernVM-FS repo is enough to also trigger a mount of the `cvmfs-config.cern.ch`?

Reply:
To clarify: accessing `/cvmfs/software.eessi.io` was working for me because we have CernVM-FS installed system-wide as well, so I guess the mount picks up on `/cvmfs/cvmfs-config.cern.ch` from the host OS.

If CernVM-FS is not installed in the host OS, then it doesn't work, so we do indeed need to explicitly mount `/cvmfs/cvmfs-config.cern.ch` via `--fusemount`.